tpu-client/src/tpu_client.rs

```diff
@@ -69,8 +69,9 @@ pub struct TpuClient<
 
 impl<P, M, C> TpuClient<P, M, C>
 where
-    P: ConnectionPool<NewConnectionConfig = C>,
-    M: ConnectionManager<ConnectionPool = P, NewConnectionConfig = C>,
+    P: ConnectionPool<NewConnectionConfig = C> + 'static,
+    M: ConnectionManager<ConnectionPool = P, NewConnectionConfig = C> + 'static,
+    C: Sync + Send + 'static,
```

Comment:
Wouldn't these `'static` constraints be too restrictive?

Reply:
They're required in order to detach the task unfortunately.

And looking at https://docs.rs/tokio/latest/tokio/task/fn.spawn.html, I'm not sure how to make the future `T: Future + Send + 'static` without the inner types being `'static` -- do you have any suggestions?

Reply:
I think this might be solved by using Arc?

Reply:
This is around the *types*, not the *instances of the types*. I did some digging around, the requirement is pretty hard for now unfortunately https://github.com/tokio-rs/tokio/issues/2385. Here's the error:

```
error[E0310]: the parameter type `C` may not live long enough
   --> tpu-client/src/nonblocking/tpu_client.rs:356:9
    |
356 | / ...   task::spawn(async move {
357 | | ...       try_send_wire_transaction(wire_transaction, &leaders, &connection_cache).a...
358 | | ...   })
    | |________^ ...so that the type `C` will meet its required lifetime bounds
    |
help: consider adding an explicit lifetime bound...
    |
344 |     C: Send + Sync + 'static,
```

And it repeats for `P` and `M`.

Here's the simplest example that shows the issue:

```
use {
    futures::future::join_all,
    std::{fmt::Debug, time::Duration},
    tokio::task::JoinHandle,
};
trait Printable: Debug {}
impl Printable for String {}
#[derive(Debug)]
struct MyStruct<T: Printable> {
    _field: T,
}
async fn print_struct<T: Printable>(instance: MyStruct<T>) {
    tokio::time::sleep(Duration::from_millis(10)).await;
    println!("{instance:?}");
}
fn make_temp_task<T: Printable + Send + Sync + 'static>(instance: MyStruct<T>) -> JoinHandle<()
> {
    tokio::task::spawn(print_struct(instance))
}
#[tokio::main]
async fn main() {
    #[derive(Debug)]
    struct MyTempStruct;
    impl Printable for MyTempStruct {}
    let tasks = (0..10).map(|_| make_temp_task(MyStruct { _field: MyTempStruct })).collect::<Ve
c<_>>();
    join_all(tasks).await;
}
```

If `make_temp_task` does not have `T: Send + Sync + 'static`, this will not work because `T` is not ensured to live long enough.

As a better option, however, I added the required bounds only for the new functions, so this way you'll only need the types to be `'static` if you're calling the `detached` version. And I was able to relax `Send + Sync` in a lot of places. Let me know what you think!